doc/man/manora.1.scd
manora(1) ["Manora 2.2.7" ["Manora Manual"]]

# NAME

*Manora* - A simple CLI & TUI tool to display, download and save man pages as PDF files.

# DESCRIPTION

Run `manora` to open a TUI menu that allows searching through local man pages, downloading man pages from <https://manned.org>, and displaying them as PDF files.

Alternatively, specify the man page to open as an argument (e.g. `manora ls`).
If the specified man page cannot be found locally, Manora offers to download it from <https://manned.org>.

Manora opens man pages in the default PDF reader defined in XDG MIME Applications, or fallback to Zathura if no default PDF reader is set.  

Manora can also be opened from a keybinding. For instance, one can bind the `alacritty -e manora` command to a keybind, which will open the `manora` TUI menu in `alacritty`.

See below for the CLI options.

# SYNOPSIS

*manora* [OPTION] [MAN_PAGE]

# OPTIONS

*-m, --menu*
	Open a TUI menu that allows searching through local man pages, downloading man pages from <https://manned.org>, and displaying them as PDF files (default operation).

*-s, --save* <man page> <file>
	Save <man page> into the <file> PDF file.
	If <file> isn't specified, save it to a "man_<man page>.pdf" file in the current directory.
	If <file> already exists, ask for a confirmation to overwrite it.

*-d, --download* <option> <man page>
	Skip searching for the man page locally and directly download it from <https://manned.org> instead.
	This option can be used when specifying a man page to open as an argument (`manora --download <man page>`) or in combination with the `-s / --save` option (`manora --download --save <man page>`).

*-V, --version*
	Display version information.

*-h, --help*
	Display the help message.

# EXIT STATUS

*0* OK.

*1* Invalid option or unknown / missing man page.

*2* Error when opening the man page as a PDF.

*3* Error when converting and exporting the man page as a PDF.

*4* Error when creating the cache directory.

*5* Error when downloading the man page from https://manned.org.

*6* Error when asking confirmation to user.

# SEE ALSO

*man*(1),
*groff*(1),
*xdg-mime*(1),
*xdg-open*(1)

# BUGS

Please report bugs to the GitHub page: https://github.com/Antiz96/manora

# AUTHOR

Robin Candau <robincandau@mailbox.org>


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

## [v2.2.7](https://github.com/Antiz96/manora/releases/tag/v2.2.7) - 2026-10-05

### Miscellaneous

- *(deps)* Lock file maintenance ([#145](https://github.com/Antiz96/manora/pull/145)) - ([f90e776](https://github.com/Antiz96/manora/commit/f90e77634a0eae2f53a7ed9961f547d9a3c7d1aa)) by @renovate[bot]
- *(deps)* Lock file maintenance ([#144](https://github.com/Antiz96/manora/pull/144)) - ([d3755af](https://github.com/Antiz96/manora/commit/d3755afcfce1fb1cc232630d9d9250a8e977eda2)) by @renovate[bot]
- *(deps)* Lock file maintenance ([#143](https://github.com/Antiz96/manora/pull/143)) - ([10b333d](https://github.com/Antiz96/manora/commit/10b333d9e2ca75eee1d60929ca8daced27643703)) by @renovate[bot]
- *(deps)* Update Rust crate clap to 4.6.7 ([#142](https://github.com/Antiz96/manora/pull/142)) - ([bfb5c9d](https://github.com/Antiz96/manora/commit/bfb5c9d429f1c7c01df32942997d307653eabaf0)) by @renovate[bot]
- *(deps)* Lock file maintenance ([#141](https://github.com/Antiz96/manora/pull/141)) - ([4f00007](https://github.com/Antiz96/manora/commit/4f00007b724bc5a30173c5372e9d032115b90dec)) by @renovate[bot]
- *(deps)* Update Rust crate reqwest to 0.13.5 ([#140](https://github.com/Antiz96/manora/pull/140)) - ([37912eb](https://github.com/Antiz96/manora/commit/37912ebfdfee52055f38dc86c12b210670f126b2)) by @renovate[bot]
- *(deps)* Lock file maintenance ([#139](https://github.com/Antiz96/manora/pull/139)) - ([47a720b](https://github.com/Antiz96/manora/commit/47a720bdcd1d2d84d3691082bb7884f1611cd69f)) by @renovate[bot]
- *(deps)* Lock file maintenance ([#137](https://github.com/Antiz96/manora/pull/137)) - ([f3fe85d](https://github.com/Antiz96/manora/commit/f3fe85d21e6d2afef84c9b46fc4f0492e83e6cd3)) by @renovate[bot]
- *(deps)* Update Rust crate which to 8.0.6 ([#136](https://github.com/Antiz96/manora/pull/136)) - ([7b8780c](https://github.com/Antiz96/manora/commit/7b8780c205a5c1bed2785d03e5d2c90db054fe98)) by @renovate[bot]
- *(deps)* Lock file maintenance ([#135](https://github.com/Antiz96/manora/pull/135)) - ([caa9ebf](https://github.com/Antiz96/manora/commit/caa9ebf06617cf20f83e246cf495d494c900b38f)) by @renovate[bot]
- *(deps)* Lock file maintenance ([#134](https://github.com/Antiz96/manora/pull/134)) - ([4a4916e](https://github.com/Antiz96/manora/commit/4a4916ed5d34bf4bba979567673eed68c6d8780c)) by @renovate[bot]
- Update email address ([#138](https://github.com/Antiz96/manora/pull/138)) - ([b96dd3f](https://github.com/Antiz96/manora/commit/b96dd3f32a4bce470061b98497e4c6d34cdd906a)) by @Antiz96

## [v2.2.6](https://github.com/Antiz96/manora/releases/tag/v2.2.6) - 2026-08-10

### Styling

- *(man)* Remove useless line breaks ([#132](https://github.com/Antiz96/manora/pull/132)) - ([cd82144](https://github.com/Antiz96/manora/commit/cd8214428806527d38a4fe0e7bbcef68c988f8fb)) by @Antiz96

### Miscellaneous

- *(deps)* Lock file maintenance ([#131](https://github.com/Antiz96/manora/pull/131)) - ([84ef9b3](https://github.com/Antiz96/manora/commit/84ef9b3318eb2e4ab833906e5dfd50d8d599bb20)) by @renovate[bot]
- Improve wording of some errors message ([#133](https://github.com/Antiz96/manora/pull/133)) - ([5e26da6](https://github.com/Antiz96/manora/commit/5e26da66c912cf1d97abaadcfdd703f1e8a2c409)) by @Antiz96

## [v2.2.5](https://github.com/Antiz96/manora/releases/tag/v2.2.5) - 2026-08-09

### Documentation

- Minor wording improvements ([#120](https://github.com/Antiz96/manora/pull/120)) - ([420c8f4](https://github.com/Antiz96/manora/commit/420c8f436184307152f936219dae4dd71c89f25d)) by @Antiz96

### Styling

- *(man)* Typo fix ([#118](https://github.com/Antiz96/manora/pull/118)) - ([191d74e](https://github.com/Antiz96/manora/commit/191d74e7eda046975d26dac9bac05e7132012427)) by @Antiz96
- Typo fix in PR template ([#119](https://github.com/Antiz96/manora/pull/119)) - ([db8e027](https://github.com/Antiz96/manora/commit/db8e0279a4abe802f133a2175b3d16ae3cf8fc23)) by @Antiz96

### Miscellaneous

- *(CI)* Add built man page to cleanup stage ([#116](https://github.com/Antiz96/manora/pull/116)) - ([32b3a13](https://github.com/Antiz96/manora/commit/32b3a13b3dec5bb477fc09940a71cded9f7a54f0)) by @Antiz96
- *(deps)* Lock file maintenance ([#130](https://github.com/Antiz96/manora/pull/130)) - ([a4af74f](https://github.com/Antiz96/manora/commit/a4af74f7c0190323df63a6613abb17228475db9d)) by @Antiz96
- *(deps)* Update Rust crate clap to 4.6.6 ([#128](https://github.com/Antiz96/manora/pull/128)) - ([9a0b527](https://github.com/Antiz96/manora/commit/9a0b527b3906f8c030a01a9c9db2227713395d3c)) by @renovate[bot]
- *(deps)* Lock file maintenance ([#124](https://github.com/Antiz96/manora/pull/124)) - ([f592e59](https://github.com/Antiz96/manora/commit/f592e592fdbe08ca21b36ac7ab3041427c76353d)) by @renovate[bot]
- *(deps)* Update Rust crate clap to 4.6.5 ([#123](https://github.com/Antiz96/manora/pull/123)) - ([86a330f](https://github.com/Antiz96/manora/commit/86a330f295b304ad9bf7d2e757646f3fb8dc0ced)) by @renovate[bot]
- *(deps)* Lock file maintenance ([#122](https://github.com/Antiz96/manora/pull/122)) - ([53ee9d6](https://github.com/Antiz96/manora/commit/53ee9d6f76be3941836dfeb40c34a86e147a923b)) by @renovate[bot]
- *(deps)* Update Rust crate clap to 4.6.4 ([#121](https://github.com/Antiz96/manora/pull/121)) - ([a8c7db9](https://github.com/Antiz96/manora/commit/a8c7db922bb28511b324b6b95b23e8537ffef957)) by @renovate[bot]
- *(deps)* Update Rust crate clap to 4.6.3 ([#117](https://github.com/Antiz96/manora/pull/117)) - ([e3ab5e3](https://github.com/Antiz96/manora/commit/e3ab5e35c5337f043cfa9c4d66bb1f515929f585)) by @renovate[bot]
- *(i18n)* Refine zathura detection ([#126](https://github.com/Antiz96/manora/pull/126)) - ([a660239](https://github.com/Antiz96/manora/commit/a66023931b6f93345fb39656a04262dbda36c461)) by @Antiz96

src/manora.sh
#!/bin/bash

# manora: A simple CLI tool to display (or save) man pages as PDFs files for an easier reading.
# https://github.com/Antiz96/manora
# Licensed under the GPL-3.0 license (or any later version of that license).

# General variables
name="manora"
version="1.3.3"
option="${1}"
man_selected="${2}"
file="${3}"

# Create tmp dir if it doesn't exists
tmpdir="${TMPDIR:-/tmp}/${name}-${UID}"
mkdir -p "${tmpdir}"

# Display debug traces if the -D / --debug argument is passed
for arg in "${@}"; do
	case "${arg}" in
		-D|--debug)
			set -x
		;;
	esac
done

# Reset the option var if it is equal to -D / --debug (to avoid false negative "invalid option" error)
# shellcheck disable=SC2154
case "${option}" in
	-D|--debug)
		unset option
	;;
esac

# Definition of the help function: Print the help message
help() {
	cat <<EOF
A simple CLI tool to display (or save) man pages as PDFs files for an easier reading.

You can directly specify the man page to display as a PDF. For instance, to display the 'ls' man page:
${name} ls

Options:
  -m, --menu                      Show a dynamic menu (via rofi or dmenu) that lists every man pages to choose from (default operation)
  -o, --output <man page> <file>  Save <man page> into the <file> PDF file
  -O, --save <man page>           Save <man page> into a "man_<man page>.pdf" file in the current directory; if <man page> isn't specified, open a menu via rofi or dmenu that lists every man pages to choose from
  -D, --debug                     Display debug traces
  -h, --help                      Display this message
  -V, --version                   Display version information

For more information, see the ${name}(1) man page.
EOF
}

# Definition of the version function: Print the current version
version() {
	echo "${name} ${version}"
}

# Definition of the invalid_option function: Print an error message, ask the user to check the help and exit
invalid_option() {
	echo -e >&2 "${name}: invalid option -- '${option}'\nTry '${name} --help' for more information"
	exit 1
}

# Definition of the check_man_page function: Check if "${man_selected}" matches a man page (used in the "print_to_pdf" and "save_to_file" functions)
check_man_page() {
	if ! man -k . | awk '{print $1}' | grep -iq ^"${man_selected}"$; then
		echo -e >&2 "${name}: man page not found -- '${man_selected}'\nTry '${name} --help' for more information"
		exit 1
	fi
}	

# Definition of the print_to_pdf function: Display "${man_selected}" in a PDF file
print_to_pdf() {
	check_man_page

	# Check if a default PDF reader is set, otherwise fallback to zathura
	if [ -n "$(xdg-mime query default application/pdf)" ]; then
		pdf_reader="xdg-open"
	elif command -v zathura > /dev/null; then
		pdf_reader="zathura"
	else
		echo -e >&2 "There is no default PDF reader defined and Zathura isn't installed\nPlease, define a default PDF reader in XDG Mime Applications or install Zathura"
		exit 2
	fi

	man -Tpdf "${man_selected}" > "${tmpdir}/${man_selected}.pdf"
	"${pdf_reader}" "${tmpdir}/${man_selected}.pdf" &
}

# Definition of the menu function: Print the list of available man pages through a `rofi` or `dmenu` menu
# --- Drop that in favor of ratatui ---
menu() {
	if command -v rofi > /dev/null; then
		man_selected=$(man -k . | awk '{print $1}' | rofi -dmenu -sort)
	elif command -v dmenu > /dev/null; then
		man_selected=$(man -k . | awk '{print $1}' | dmenu -l 15)
	else
		echo -e >&2 "A dynamic menu is required to print the list of available man pages\nPlease, install rofi or dmenu"
		exit 2
	fi
}

# Definition of the save_to_file function: Save output to "${file}" (used in the "output" and "save" functions) 
save_to_file() {
	check_man_page

	if [ -f "${file}" ]; then
		echo "The ${file} file already exists"
		read -rp $'Do you want to overwrite its content? [y/N] ' answer

		case "${answer}" in
			[Yy])
				echo
			;;
			*)
				echo -e >&2 "\nAborting"
				exit 3
			;;
		esac
	fi

	man -Tpdf "${man_selected}" > "${file}"
	echo "The ${man_selected} man page has been saved to the ${file} file"
}

# Definition of the output function: Save the man page to the specified PDF file
output() {
	if [ -z "${man_selected}" ] || [ -z "${file}" ]; then
		echo -e >&2 "Please, specify a man page to export and a file to save it to: ${name} -o man_page /path/to/file\nTry '${name} --help' for more information"
		exit 3
	fi

	save_to_file
}

# Definition of the save function: Save the man page to a PDF file named "man_<command>.pdf" in the current directory
save() {
	if [ -z "${man_selected}" ]; then
		menu
	fi
	
	file="man_${man_selected}.pdf"

	save_to_file
}


# Execute the different functions depending on the option
case "${option}" in
	-m|--menu|"")
		menu && print_to_pdf
	;;
	-o|--output)
		output
	;;
	-O|--save)
		save
	;;
	-h|--help)
		help
	;;
	-V|--version)
		version
	;;
	*)
		if [ "${option:0:1}" == "-" ]; then
			invalid_option
		else
			man_selected="${option}"
			print_to_pdf
		fi
	;;
esac
